k4FWCore/include/k4FWCore/Consumer.h

```diff
@@ -90,6 +91,13 @@ namespace k4FWCore {
 
       // ... instead, they must implement the following operator
       virtual void operator()(const In&...) const = 0;
+
+      auto inputLocations(int i) const {
+        if (i >= sizeof...(In)) {
+          throw GaudiException("Called inputLocations with an index out of range", "Consumer", StatusCode::FAILURE);
+        }
+        return m_inputLocations[i] | std::views::transform([](const DataObjID& id) -> const auto& { return id.key(); });
```

Comment:
This will require a change here:

https://github.com/key4hep/k4FWCore/blob/beb9affb476f97f313a6d0ba20f914ea86100a61/CMakeLists.txt#L42-L46

Reply:
Indeed, actually this should have been changed earlier. For spack it's fine and locally I set it to C++20, if no one complained, is no one building k4FWCore? ...

Reply:
I am, but also with c++20 :sweat_smile: